kafka-connect-transforms/README.md

```diff
@@ -47,3 +47,72 @@ It will promote the `before` or `after` element fields to top level and add the
 | Property            | Description                                                                       |
 |---------------------|-----------------------------------------------------------------------------------|
 | cdc.target.pattern  | Pattern to use for setting the CDC target field value, default is `{db}.{table}`  |
+
+# JsonToMapTransform
+_(Experimental)_ 
+
+The `JsonToMapTransform` SMT parses Json object payloads.  It is intended for use when the incoming data is too 
+inconsistent for the `iceberg-connector` to infer `Struct` schemas from (e.g. the object keys are too variable 
+and this leads to an explosion of schema evolutions and columns).  This will get the data into Iceberg where it can 
+be further processed by query engines into a more manageable form.  
+
+It will parse string values into `SinkRecords` with a Schema and a Struct.  It assumes the messages are json objects 
+themselves and will throw exceptions if the records are primitives.  You must use `StringConverter` as the 
+`ValueConverter`, not `org.apache.kafka.connect.json.JsonConverter`.
+
+Keys are not transformed. 
```

Comment:
meh, my attempt at wording: 

```
The default behaviour of the iceberg-kafka-connect connector for schema-less data is to infer Maps as Iceberg Struct 
types (rather than Map type). This is usually fine when working with well-structured data but when you have JSON objects 
with large numbers of dynamically changing keys, it can lead to an explosion of columns in your Iceberg table.

The `JsonToMapTransform` SMT is useful in these types of situations as it parses (string) JSON objects into `SinkRecords` 
with a `Schema` that represents JSON objects as a Map type (rather than Struct type). The connector will respect the `Schema`
and create Iceberg tables with Iceberg Map type columns for the JSON objects.  

Note:
- The SMT accepts only messages with String values as valid input. If this SMT is reading data directly from Kafka, we 
  recommend using the `StringConverter` as the `valueConverter` setting for your connector rather than
  `org.apache.kafka.connect.json.JsonConverter`. 
- In addition, the strings must represent JSON objects, otherwise it will throw an exception.
- Message keys are not transformed but passed along as-is by the SMT.
```